connmgr/connmanager.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2016 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package connmgr
+
+import (
+	"fmt"
+	mrand "math/rand"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/addrmgr"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/wire"
+)
+
+var (
+	//MaxOutboundPeers is the number of max outbound connections.
+	MaxOutboundPeers = 8
```

Comment:
All of these need to be part of a `Config` struct that is provided when creating a new connmgr instance instead of globals.